src/vehicle.cpp

```diff
@@ -3193,6 +3216,42 @@ int vehicle::basic_consumption(const itype_id &ftype) const
     return fcon;
 }
 
+int vehicle::supplemental_consumption( const itype_id &ftype ) const
+{
+    int fcon = 0;
+    if( chainsaw_on ) {
+        std::vector<int> all_chainsaws = all_parts_with_feature( "CHAINSAW" );
+        for( size_t c = 0; c < all_chainsaws.size(); ++c ) {
+            if( part_info( all_chainsaws[c] ).fuel_type == ftype ) {
+                fcon += part_power( all_chainsaws[c] );
+                tripoint part_pos = global_pos3() + parts[ all_chainsaws[c] ].precalc[0];
+                // show onomatopoeia rarely but emit noise all time
+                //~ sound of working chainsaw
+                sounds::sound( part_pos, 20, one_in( 13 ) ? _( "brrr!" ) : "" );
+            }
+        }
+    }
+    return fcon;
+}
+
+void vehicle::disable_devices( const itype_id &ftype )
+{
+    //fixme: running out of any fuel type used by chainsaw(s) disables all chainsaws
+    if( chainsaw_on ) {
+        std::vector<int> all_chainsaws = all_parts_with_feature( "CHAINSAW" );
+        for( size_t c = 0; c < all_chainsaws.size(); ++c ) {
+            if( part_info( all_chainsaws[c] ).fuel_type == ftype ) {
+                chainsaw_on = false;
+                if( player_in_control( g->u ) || g->u.sees( global_pos3() ) ) {
+                     //~ %1s is vehicle name and %2s is fuel type
+                     add_msg( _( "The %1$s's ran out of %2$s!" ), name.c_str(),  ftype.c_str() );
```

Comment:
The fuel type must be translated to the fuel name (see for example in `vehicle::start_engine`), something like `item::nname( ftype )` instead of `ftype` should do it.